cmd/kes/log.go

```diff
@@ -79,60 +93,200 @@ func logTrace(args []string) error {
 	if err != nil {
 		return err
 	}
-	stream, err := client.TraceAuditLog()
-	if err != nil {
+
+	switch strings.ToLower(typeFlag) {
+	case "audit":
+		stream, err := client.TraceAuditLog()
+		if err != nil {
+			return err
+		}
+		defer stream.Close()
+
+		if !isTerm(os.Stdout) || jsonOutput {
+			closeOn(stream, os.Interrupt, os.Kill)
+			for stream.Next() {
+				fmt.Println(string(stream.Bytes()))
+			}
+			return stream.Err()
+		}
+		return traceAuditLogWithUI(stream)
+	case "error":
+		stream, err := client.TraceErrorLog()
+		if err != nil {
+			return err
+		}
+		defer stream.Close()
+
+		if !isTerm(os.Stdout) || jsonOutput {
+			closeOn(stream, os.Interrupt, os.Kill)
+			for stream.Next() {
+				fmt.Println(string(stream.Bytes()))
+			}
+			return stream.Err()
+		}
+		return traceErrorLogWithUI(stream)
+	default:
+		return fmt.Errorf("Unknown log event type: '%s'", typeFlag)
+	}
+}
+
+// traceAuditLogWithUI iterates over the audit log
+// event stream and prints a table-like UI to STDOUT.
+//
+// Each event is displayed as a new row and the UI is
+// automatically adjusted to the terminal window size.
+func traceAuditLogWithUI(stream *kes.AuditStream) error {
+	table := xterm.NewTable("Time", "Identity", "Status", "API Operations", "Response")
+	table.Header()[0].Width = 0.12
```

Comment:
could `table.Header()` be `empty` or `nil` at some point? that will cause `npe`

Reply:
It can an `empty` slice. However, it's kind of the callers responsibility to access only table headers that exists.
Here I define 5 columns and therefore 5 headers exists:
```
table := xterm.NewTable("Time", "Identity", "Status", "API Operations", "Response")
```
See: https://github.com/minio/kes/pull/68/files#diff-9e123ecf3fdb4c1d6991d8cb7b908636R71-R72